PostHog/PostHogStorage.swift

```diff
@@ -46,7 +83,8 @@ class PostHogStorage {
     init(_ config: PostHogConfig) {
         appFolderUrl = Self.getAppFolderUrl(from: config)
 
-        createDirectoryAtURLIfNeeded(url: appFolderUrl)
+        // migrate legacy storage if needed
+        Self.migrateLegacyStorage(from: config, to: appFolderUrl)
```

Comment:
should this be done after the SDK is init?
just wondering if `config` is needed but the values set are not fully started yet because this is executed in the ctor before the SDK is fully functional